### DISCUSSION TITLE
Renaming server components

### DISCUSSION THREAD

**Discussion by nobkd** [nuejs/nue#382] [General]
Hey @tipiirai (and everyone else), 

I open this discussion, because I find the naming of *server components* in nue really unintuitive:

If I read `server (side) component`, I always think of components that are (re-)rendered on a (js-)server on **runtime**, but nue's *"server components"* run on **compile- / build-time**.

What do you think about using the term `layout component` or `build- / compile-time component` instead?
I find that much more descriptive, of what they really are, and users won't expect them to run on a js server and recompute using changing values e.g. from a database.

What do you think? 

Related:
- #238
- #251
- #231
- (#234)

**Comment by seneca** [comment]
It's tricky to comment on that because I don't know what's in the mind of the author, in other words, how he sees server components to work in nue. A HTML-component or nue-component can be thrown into the hat as another option for a name.

**Reply by nobkd** [reply]
Thanks for your comment. 

I personally don't agree to use the name `nue-component` for server components. It may be correct, but at least in my opinion, layout components (server components) and client side components are both nue components (See code block for idea), just the one renders to html on build time, and the other one gets rendered (and interactive) on client side.

Sure, HTML-component as name might work. 

---

```
-- Nue components
   |
   |-- client side components
   |-- layout components
```

**Comment by tipiirai** [comment]
I hear you. I think these are the alternatives and their benefits:

* Server components (The general term)
* Layout components (Exact term for layout components)
* Server islands (Familiar Astro terminology)

Some server-side components may support the layout "slots", like for example an `<author/>` component.

**Reply by tipiirai** [reply]
I think "Layout components" are good, since `<author/>` is also layout. Server islands _might_ be good marketing-wise.

**Reply by nobkd** [reply]
Okay. Thanks for explaining. I didn't use Astro, so I didn't know, they used this term. I understand your naming better now :)

**Comment by tipiirai** [comment]
Here's the new planned table of contents for the documentation area. Feedback is welcome! I want to nail this. 

<img width="283" alt="image" src="https://github.com/user-attachments/assets/01132aed-ac71-4586-b05a-d59e5769df92">

**Reply by seneca** [reply]
I think it would be great to have a Step-by-Step tutorial in the documentation area. Quick start is great for those who are somehow familiar with Web Development but having a tutorial is great for those who want to learn by doing.

**Reply by tipiirai** [reply]
The "Quick start" is actually the step-by-step tutorial. It was called "Tutorial" before, but then a friend of mine suggested to rename it to "Quick start". Would you prefer "Tutorial" or perhaps even "Step-by-step tutorial"?

**Comment by seneca** [comment]
I understand what you are saying, however, if you follow the Quck Start you will not arrive at the end product as shown on the website. A lot of steps are assumed and no files are available for download to follow along. By a Step-by-Step tutorial I really mean do this first, then do that next, etc. Of course Quick Start can be updated to become a Step-by-Step tutorial. I completely understand that you are up to your ears with something else, but for the new release it would be great to have it upgraded.
